tests/server/mod.rs

```diff
@@ -89,13 +125,36 @@ where
     F: Fn(Request<hyper::body::Incoming>) -> Fut + Send + Sync + 'static,
     Fut: Future<Output = Response<Body>> + Send + 'static,
 {
-    http_inner(Arc::new(move |req| Box::new(Box::pin(func(req)))))
+    http_inner(Arc::new(move |req| Box::new(Box::pin(func(req)))), None)
+}
+
+#[cfg(unix)]
+pub fn http_unix<F, Fut>(func: F) -> Server
+where
+    F: Fn(Request<hyper::body::Incoming>) -> Fut + Send + Sync + 'static,
+    Fut: Future<Output = Response<Body>> + Send + 'static,
+{
+    use rand::Rng;
+    let file_name: String = rand::thread_rng()
+        .sample_iter(&rand::distributions::Alphanumeric)
+        .take(10)
+        .map(char::from)
+        .collect();
```

Comment:
Looks like `tempfile` can be used for this: https://docs.rs/tempfile/latest/tempfile/struct.Builder.html#examples-11

That way we also get cleanup